### DISCUSSION TITLE
Are there instructions for setting up Apprise?

### DISCUSSION THREAD

**Discussion by nbanderso** [mcguirepr89/BirdNET-Pi#337] [General]
I’ve looked all over but haven’t been able to find instructions on setting up apprise.  Can someone walk me through this?

Thank you!

**Comment by ehpersonal38** [comment]
Hi there!

First you need to know what notification service you want to use with Apprise. Apprise is basically the "man in the middle" that allows our notifications to reach your service. So, go to this link: https://github.com/caronc/apprise/wiki, and scroll down until you find the list of 74 services.

Then you can click on the service you're interested in. For example, if you want notifications to be posted to Discord, click Discord. If you want Telegram, click Telegram. Then follow the helpful guide that Apprise provides to put together your URL string. This usually involves logging in to your service and accessing some "Developer" area in the settings to see your API keys.

You will end up creating a URL string that looks a bit like this:
`https://discordapp.com/api/webhooks/4174216298/JHMHI8qBe7bk2ZwO5U711o3dV_js`
but obviously your service will be different than discordapp.com.

Finally, in BirdNET-Pi, go to Tools -> Settings, and paste your URL string into the large text area "Apprise Notifications Configuration:". A few lines below, check each desired notification option, like "Notify each new detection," and you should be on your way.

Hope this helps!
🐤

**Reply by nbanderso** [reply]
Thanks for the great explanation!  It looks like I am having some issues with the tools section of Birdnet-Pi.  

The only relevant option that I have in the tools section is for Pushed App.  I’ve searched the entire interface and there is no reference to Apprise.  Any ideas on how to add that?

Thanks again!
![F97CBA89-CBE9-44D3-9B58-97F8ACB3DE44](https://user-images.githubusercontent.com/52547554/171194683-f5a23f08-70b5-41a5-9e5a-e4ae690f4e1d.jpeg)

**Reply by ehpersonal38** [reply]
You're definitely out of date there! Head to Tools -> System Controls -> Update. It will take quite a while, and you might need to do it twice, but it should work!

**Reply by nbanderso** [reply]
Thank you!  

Just for reference, I setup birdnet-Pi last week following the instructions on birdnetpi.com.  More specifically, I pasted the curl command into my raspberry pi's command prompt. 

Not sure how I ended up with an old version.

**Reply by ehpersonal38** [reply]
Hmm, things do move fast around here but Apprise was added about 3 weeks ago. Let me know if the update doesn't work though, you may be on a different repo

**Comment by ehpersonal38** [comment]
I will mention that Google [has discontinued support for less-secure apps as of yesterday.](https://support.google.com/accounts/answer/6010255?hl=en#:~:text=To%20help%20keep%20your%20account,only%20your%20username%20and%20password.) This means that setup for Gmail with Apprise is now a little more difficult. The workaround that I have found is to [enable 2-Step Vertification](https://support.google.com/accounts/answer/185839?hl=en&co=GENIE.Platform%3DDesktop) on your Gmail account, then [generate an App Password](https://support.google.com/accounts/answer/185833?hl=en). Then you can use this app password in place of your usual Gmail password in the E-mail URL string for Apprise:
`mailto://{user}:{your_app_password}@gmail.com`

**Reply by blogdemoi** [reply]
And now I can be notified using my custom e-mail, with:

mailtos://arnauld:psswd@arnauld.org:587?smtp=arnauld.org
